### PR TITLE
Add collection request spec and reorganize request specs for consistency

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made?

To add some test cases.

## How was this change tested?

By running CI.

## Which documentation and/or configurations were updated?

None

### CODE REVIEW

File: spec/requests/authenticate_user_spec.rb [hunk 1]
```diff
@@ -44,6 +44,12 @@
       get "/works/#{work.id}"
       expect(response).to have_http_status(:success)
     end
+
+    it 'redirects from /collections/new to login URL' do
+      get '/collections/new'
```
Comment (jcoyne) [inline spec/requests/authenticate_user_spec.rb:49]: Should these go in the collections spec?

File: spec/requests/collections_creation_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    it 'creates a new collection' do
+      post '/collections', params: collection_params
+      expect(response).to have_http_status(:found)
+      expect(response).to redirect_to(dashboard_path)
+    end
+  end
+
+  # TODO: Uncomment when Collection validations are added
```
Comment (jcoyne) [inline spec/requests/collections_creation_spec.rb:33]: They are added, do a rebase?

File: spec/requests/authenticate_user_spec.rb (hunk 1 above)
Comment (mjgiarlo) [reply spec/requests/authenticate_user_spec.rb:49]: I could go either way, but the current pattern concentrates all authN stuff into this request spec.

File: spec/requests/collections_creation_spec.rb (hunk 2 above)
Comment (mjgiarlo) [reply spec/requests/collections_creation_spec.rb]: Done.

**Comment by mjgiarlo** [comment]
@jcoyne have another look?

File: spec/requests/authenticate_user_spec.rb (hunk 1 above)
Comment (jcoyne) [reply spec/requests/authenticate_user_spec.rb:49]: I think we should lay these out per request (rather than per-feature).

Comment (mjgiarlo) [reply spec/requests/authenticate_user_spec.rb:49]: @jcoyne K. I'll re-organize in the morning.

Comment (mjgiarlo) [reply spec/requests/authenticate_user_spec.rb]: @jcoyne Done. This is ready for another look now. The request specs now line up better w/ the controllers.

File: spec/requests/login_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# typed: false
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+# NOTE: in deployed environments, this interaction is handled by Apache and
+#       Shibboleth in concert with the application
+RSpec.describe 'Login/out requests' do
```
Comment (jcoyne) [inline spec/requests/login_spec.rb:8]: Can we delete spec/controllers/ now?

Comment (mjgiarlo) [reply spec/requests/login_spec.rb:8]: No, I still need to try to convert that guy to a feature spec to test the "user return to" logic. (I'll do that next.)

**Review by jcoyne — APPROVED** [review]

**Pull request merged**